### PR TITLE
feat: LT-2310 - manage routing rules for all teams

### PR DESCRIPTION
**Pull request by Mark-Spiers-kainos**

### CODE REVIEW

File: assets/css/components/_routing-rules.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.deactivated {
+	opacity: 0.65;
+	margin-bottom: - govuk-spacing(4);
+}
```
Comment (andytaylorkainos) [inline assets/css/components/_routing-rules.scss:4]: new line pls

File: routing_rules/views.py [hunk 2]
```diff
@@ -73,10 +73,20 @@ def get(self, request, **kwargs):
 
 class CreateRoutingRule(MultiFormView):
     def init(self, request, **kwargs):
+        select_team = has_permission(request, Permission.MANAGE_ALL_ROUTING_RULES)
         if request.method == "POST":
-            self.forms = routing_rule_form_group(request, request.POST.getlist("additional_rules[]"))
+            team_id = request.POST.get("team", get_gov_user(request)[0]["user"]["team"]["id"])
```
Comment (andytaylorkainos) [inline routing_rules/views.py:78]: can you make this a service, please?

File: assets/css/components/_routing-rules.scss [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.deactivated {
```
Comment (jfaracik) [inline assets/css/components/_routing-rules.scss:1]: This should follow BEM (http://getbem.com) - Something like `.app-routing-rule__item--deactivated`

Comment (jfaracik) [reply assets/css/components/_routing-rules.scss:1]: Also a new line at the end of the file :)

File: conf/context_processors.py [hunk 4]
```diff
@@ -58,7 +58,8 @@ def lite_menu(request):
                 {"title": "Flagging rules", "url": reverse_lazy("flags:flagging_rules"), "icon": "menu/flags"},
             ),
             conditional(
-                has_permission(request, Permission.MANAGE_TEAM_ROUTING_RULES),
+                has_permission(request, Permission.MANAGE_TEAM_ROUTING_RULES)
```
Comment (jfaracik) [inline conf/context_processors.py:61]: We should be careful about calling has_permission multiple times in this file, I believe it does a request to the API for each one. Perhaps get all permissions on this page and then assert that the permission is in the list.

File: ui_automation_tests/pages/routing_rules_pages.py [hunk 5]
```diff
@@ -19,6 +19,7 @@ class RoutingRulesPage(BasePage):
     SELECT_FLAG_ID = "flag"
     CHECKBOX_AND_LABEL = ".govuk-checkboxes__item"
     RADIO_BUTTONS = ".govuk-radios__input"
+    TEAM_PREFIX_ID = "team-"
```
Comment (jfaracik) [inline ui_automation_tests/pages/routing_rules_pages.py:22]: TEAM_ID_PREFIX

File: templates/routing-rules/index.html [hunk 6]
```diff
@@ -63,7 +63,11 @@ <h1 class="govuk-heading-l">
 						<td class="govuk-table__cell">
 							{% if rule.flags %}
 								{% for flag in rule.flags %}
-									{{ flag.name }}
+									{% if flag.status == "Active" %}
```
Comment (jfaracik) [inline templates/routing-rules/index.html:66]: I'd be worried about doing an equal on a presentation value, does flag.status not have a key would compare to?

File: teams/services.py [hunk 7]
```diff
@@ -45,3 +45,10 @@ def get_users_by_team(request, pk, convert_to_options=False):
     if convert_to_options:
         return [Option(user["id"], user["email"]) for user in data.json()["users"] if user["email"]]
     return data.json(), data.status_code
+
+
+def get_team_queues(request, pk, convert_to_option=False):
+    data = get(request, TEAMS_URL + pk + "/queues/")
+    if convert_to_option:
```
Comment (jfaracik) [inline teams/services.py:52]: `convert_to_options`

And then only return `return data.json()`, we dont return status codes for gets

File: routing_rules/views.py (hunk 2 above)
Comment (jfaracik) [reply routing_rules/views.py:78]: How would you make that into a service?

Comment (jfaracik) [reply routing_rules/views.py:78]: Could something like this work?

```
class CreateRoutingRule(MultiFormView):
    def init(self, request, **kwargs):
        select_team = has_permission(request, Permission.MANAGE_ALL_ROUTING_RULES)
        team_id = request.POST.get("team", get_gov_user(request)[0]["user"]["team"]["id"])
        self.forms = routing_rule_form_group(
                request=request,
                additional_rules=request.POST.getlist("additional_rules[]", ()),
                team_id=team_id,
                select_team=select_team,
        )
        self.success_url = reverse("routing_rules:list")
        self.action = post_routing_rule
```

File: templates/routing-rules/index.html (hunk 6 above)
Comment (Mark-Spiers-kainos) [reply templates/routing-rules/index.html:66]: Flag status is returned as a string just

File: routing_rules/views.py (hunk 2 above)
Comment (Mark-Spiers-kainos) [reply routing_rules/views.py]: That looks like it works well.

Comment (andytaylorkainos) [reply routing_rules/views.py]: > How would you make that into a service?

Woops - misread the line: assumed it was doing a POST

**Pull request merged**